design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,212 @@
+type Mutation {
+    createNegotiableQuote(
+        cartId: ID! @doc(description: "Cart ID")
+        quoteName: String @doc(description: "Quote name")
+        comment: String @doc(description: "Comment")
+        files: [FileInput!] @doc(description: "Attached files")
+    ): CreateNegotiableQuoteOutput! @doc(description:"Generates negotiable quote request")
+
+    addNegotiableQuoteComments(
+        cartId: ID! @doc(description: "Cart ID")
+        comment: String @doc(description: "Comment")
+        files: [FileInput!] @doc(description: "Attached files")
+    ): AddNegotiableQuoteCommentsOutput! @doc(description:"Generates negotiable quote request")
+
+    updateNegotiableQuoteItemsQuantity(
+        cart_id: ID!, @doc(description: "unique Id of negotiable quote")
+        items: [ID!]! @doc(description: "unique Ids of Items to be update in negotiable quote")
+        quantity: Float, @doc(description: "quantity of the item  to be updated")
+    ): UpdateNegotiableQuoteItemsQuantityOutput @doc(description: "Remove Items in negotiable quote")
+
+    removeNegotiableQuoteItems(
+        cart_id: ID!, @doc(description: "unique Id of negotiable quote")
+        items: [ID!]! @doc(description: "unique Ids of Items to be removed from negotiable quote")
+    ): RemoveNegotiableQuoteItemsOutput @doc(description: "Remove Items in negotiable quote")
+
+    setShippingAddressesOnNegotiableQuote(
+        cart_id: ID!
+        shipping_addresses: [NegotiableQuoteShippingAddressInput!]!
+    ): SetShippingAddressesOnNegotiableQuoteOutput @doc(description: "Assign shipping address to Negotiable quote")
+
+    setShippingMethodsOnNegotiableQuote(
+        cartId: ID! @doc(description: "Cart ID")
+        shippingMethods: [ShippingMethodInput]! @doc(description: "Input for shipping method")
+    ): SetShippingMethodsOnNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+}
+
+type Customer {
+    negotiable_quote (
+        id: ID!): NegotiableQuote @doc(description: "Get negotiable quote of a customer")
+
+    negotiable_quotes(
+        filter: NegotiableQuoteFilterInput,
+        pageSize: Int = 20,
+        currentPage: Int = 1
+    ): [NegotiableQuote] @doc(description: "Get negotiable quotes of a customer")
+
+    estimateShippingCostForNegotiableQuote(
```

Comment:
The `estimate` prefix for these queries isn't ideal - we want querying to feel like you're selecting data, not necessarily triggering some computation. I would expect something like a `shipping_costs` field on a type like `NegotiableQuote`.

In general, `estimateShippingCostForNegotiableQuote` seems like it's in the wrong place. It's data the UI will want when rendering a quote, and should be in the schema along with the quote,

Reply:
Hi @DrewML , removed `estimateShippingCostForNegotiableQuote` and `estimateShippingCostByAddressIdForNegotiableQuote` query, the `NegotiableQuote` is already having a field `available_shipping_methods` which gives the same output.

Thanks.